[RFC]vim-patch:8.1.{495,505,531,533,583,623,630,641,686,715,833,1012,1221,1651}

janlazo opened the pull request:
It's been RDY for a while. I'll just continue pushing trivial or independent vim patches in this PR until this gets merged or Vim releases 8.2.0000.

~~Going to stop for now. I can't trust the CI since https://github.com/neovim/neovim/commit/47d53d66bdcf1df1688c9af6999156a6c97f5aa7 .~~ Resolved quickly. Thanks.

janlazo committed:
vim-patch:8.1.0495: :filter only supports some commands

Problem:    :filter only supports some commands.
Solution:   Add :filter support for more commands. (Marcin Szamotulski,
            closes vim/vim#2856)
https://github.com/vim/vim/commit/f86db78fed78541cefdb706e4779ce5ae9ca7820

janlazo committed:
vim-patch:8.1.0505: filter command test may fail if helplang is not set

Problem:    Filter command test may fail if helplang is not set.
Solution:   Set 'helplang' for the test. (James McCoy, closes vim/vim#3591)
https://github.com/vim/vim/commit/bd9a0c611ce08f8dce033537bc2f110987b99802

janlazo committed:
vim-patch:8.1.1221: filtering does not work when listing marks

Problem:    Filtering does not work when listing marks.
Solution:   Implement filtering marks. (Marcin Szamotulski, closes vim/vim#3895)
https://github.com/vim/vim/commit/ad6dc49a7564a99fca36c1928e3865787d3bd5b2

janlazo commented:
`TermClose` is flaky now?
```
[  ERROR   ] 1 error, listed below:
[  ERROR   ] ...im/neovim/test/functional/autocmd/termclose_spec.lua @ 20: TermClose event triggers when fast-exiting terminal job stops
test/functional/helpers.lua:96: Vim(call):E900: Invalid channel id
stack traceback:
	test/functional/helpers.lua:96: in function 'request'
	test/functional/helpers.lua:254: in function 'command'
	...im/neovim/test/functional/autocmd/termclose_spec.lua:23: in function <...im/neovim/test/functional/autocmd/termclose_spec.lua:20>
 17 SKIPPED TESTS
 1 ERROR
-- Output to stderr:
CMake Error at /home/travis/build/neovim/neovim/cmake/RunTests.cmake:63 (message):
  functional tests failed with error: 1
```

Am I missing a `cmdline` patch?
```
Direct leak of 9 byte(s) in 1 object(s) allocated from:
    #0 0x50ae38 in malloc (/home/travis/build/neovim/neovim/build/bin/nvim+0x50ae38)
    #1 0x1134874 in try_malloc /home/travis/build/neovim/neovim/build/../src/nvim/memory.c:70:15
    #2 0x1134a34 in xmalloc /home/travis/build/neovim/neovim/build/../src/nvim/memory.c:104:15
    #3 0x144ec95 in home_replace_save /home/travis/build/neovim/neovim/build/../src/nvim/os/env.c:953:17
    #4 0x756d8f in buflist_nr2name /home/travis/build/neovim/neovim/build/../src/nvim/buffer.c:2396:10
    #5 0x108a331 in fm_getname /home/travis/build/neovim/neovim/build/../src/nvim/mark.c:587:10
    #6 0x1090b11 in ex_jumps /home/travis/build/neovim/neovim/build/../src/nvim/mark.c:791:14
    #7 0xc92c3e in do_one_cmd /home/travis/build/neovim/neovim/build/../src/nvim/ex_docmd.c:2248:5
    #8 0xc70ee0 in do_cmdline /home/travis/build/neovim/neovim/build/../src/nvim/ex_docmd.c:593:20
    #9 0xc771f5 in do_cmdline_cmd /home/travis/build/neovim/neovim/build/../src/nvim/ex_docmd.c:270:10
    #10 0x9ed571 in f_execute /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:8508:5
    #11 0x906917 in call_func /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:6562:11
    #12 0x91a42e in get_func_tv /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:6303:11
    #13 0x9afdf1 in eval7 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:4406:15
    #14 0x9ac393 in eval6 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:4103:7
    #15 0x9a9b07 in eval5 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3951:7
    #16 0x9a4e0f in eval4 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3687:7
    #17 0x9a42ec in eval3 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3605:7
    #18 0x9a37bc in eval2 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3536:7
    #19 0x8fecb2 in eval1 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3463:7
    #20 0x919d1d in get_func_tv /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:6276:9
    #21 0x9afdf1 in eval7 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:4406:15
    #22 0x9ac393 in eval6 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:4103:7
    #23 0x9a9b07 in eval5 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3951:7
    #24 0x9a4e0f in eval4 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3687:7
    #25 0x9a42ec in eval3 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3605:7
    #26 0x9a37bc in eval2 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3536:7
    #27 0x8fecb2 in eval1 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3463:7
    #28 0x8fe0ca in eval0 /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:3423:9
    #29 0x90b022 in ex_let_const /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:1604:9
```

erw7 commented:
> Am I missing a cmdline patch?

@janlazo You need https://github.com/vim/vim/commit/d93090f41f70c521cfad5b25efcb0024b9480082.

janlazo committed:
vim-patch:8.1.0833: memory leak when jumps output is filtered

Problem:    Memory leak when jumps output is filtered.
Solution:   Free the filtered name. (Dominique Pelle, closes vim/vim#3869)
https://github.com/vim/vim/commit/d93090f41f70c521cfad5b25efcb0024b9480082

janlazo committed:
vim-patch:8.1.1012: memory leak with E461

Problem:    Memory leak with E461.
Solution:   Clear the typeval. (Dominique Pelle, closes vim/vim#4111)
https://github.com/vim/vim/commit/ab89d7ab89a1dd7e40cc28df96c71c11b5ab9089

janlazo committed:
vim-patch:8.1.0531: flaky tests often fail with a common error message

Problem:    Flaky tests often fail with a common error message.
Solution:   Add a pattern to match an error message indicating a flaky test.
https://github.com/vim/vim/commit/dbc0d2163aa5e090d5a0c83aea448803ddbab664

janlazo committed:
vim-patch:8.1.0533: screendump tests can be flaky

Problem:    Screendump tests can be flaky.
Solution:   Add VerifyScreenDump to the pattern of flaky tests.
https://github.com/vim/vim/commit/447f6ce8bd42fe3adbdf97eff9a38a9bfef9eeed

janlazo committed:
vim-patch:8.1.1651: suspend test is flaky on some systems

Problem:    Suspend test is flaky on some systems.
Solution:   Wait for the shell prompt to show. (Yee Cheng Chin, closes vim/vim#4632)
https://github.com/vim/vim/commit/999224422633935eaa8b804ce3d1c2f9ca195d0a

janlazo committed:
vim-patch:8.1.0583: using illogical name for get_dict_number()/get_dict_string()

Problem:    Using illogical name for get_dict_number()/get_dict_string().
Solution:   Rename to start with dict_.
https://github.com/vim/vim/commit/8f66717a1f835b8194139d158c1e2df8b30c3ef3

blueyed commented on src/nvim/testdir/shared.vim:
```diff
@@ -262,6 +266,12 @@ func! Screenline(lnum)
+    let text .= item.chars
+  endfor
+  return text
 endfunc
```
We should not add `Get_terminal_text` (unused, and removed in Vim via "patch 8.1.0769: :stop is covered in two tests" later).  Merged already via https://github.com/neovim/neovim/pull/10157.

janlazo replied:
It's harmless and patch 8.1.1581 will remove it.

janlazo committed:
vim-patch:8.1.0623: iterating through window frames is repeated

Problem:    Iterating through window frames is repeated.
Solution:   Define FOR_ALL_FRAMES. (Yegappan Lakshmanan)
https://github.com/vim/vim/commit/3d1491ed2394b3e92902102879bace28a5f9c201

janlazo committed:
vim-patch:8.1.0630: "wincmd p" does not work after using an autocmd window

Problem:   "wincmd p" does not work after using an autocmd window.
Solution:   Store "prevwin" in aco_save_T. (Christian Brabandt, closes vim/vim#3690)
https://github.com/vim/vim/commit/a42df5934bdc1178ed2ee8cb9c8686975b578497

janlazo committed:
vim-patch:8.1.0641: no check for out-of-memory when converting regexp

Problem:    No check for out-of-memory when converting regexp.
Solution:   Bail out when lalloc() returns NULL. (John Marriott)
https://github.com/vim/vim/commit/c57463c9c6ee893285f553e0ac3b2fe5935f16b8

janlazo committed:
vim-patch:8.1.0686: when 'y' is in 'cpoptions' yanking for the clipboard changes redo

Problem:    When 'y' is in 'cpoptions' yanking for the clipboard changes redo.
Solution:   Do not use the 'y' flag when "gui_yank" is TRUE. (Andy Massimino,
            closes vim/vim#3760)
https://github.com/vim/vim/commit/5823f84dd04198994e3e5f2e278a5e315c47d32d

justinmk commented:
reopen?

janlazo committed:
vim-patch:8.1.0715: superfluous call to redraw_win_later()

Problem:    Superfluous call to redraw_win_later().
Solution:   Remove the call.
https://github.com/vim/vim/commit/6f7e555f7440df148350468ad8bc6d559d676d7c

justinmk commented:
Thanks @janlazo !

janlazo commented:
I forgot to submit the patch for 8.1.0715 :( . I'll followup in another PR.
